File: sieve.py
# Python
def sieve_of_eratosthenes(n):
    # Initialize all elements of boolean array `prime[0..n]` with true.
    prime = [True for i in range(n+1)]
    p = 2
    while p * p <= n:
        if prime[p]:
            # Update all multiples of p to false, i.e. not prime
            for i in range(p * p, n+1, p):
                prime[i] = False
        p += 1
 
    # Print all prime numbers
    for p in range(2, n+1):
        if prime[p]:
            print(p)
 
if __name__ == '__main__':
    sieve_of_eratosthenes(20)

File: sieve.c
// C
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

void sieveOfEratosthenes(int n) {
    // Initialize all elements of boolean array `prime[0..n]` with true.
    bool prime[n + 1];
    memset(prime, true, sizeof(prime));

    for (int p = 2; p * p <= n; p++) {
        if (prime[p]) {
            // Update all multiples of p to false, i.e. not prime
            for (int i = p * p; i <= n; i += p)
                prime[i] = false;
        }
    }

    // Print all prime numbers
    for (int p = 2; p <= n; p++)
        if (prime[p])
            printf("%d \n",p);
}

int main() {
    sieveOfEratosthenes(20);
    return 0;
}
